landlab/utils/distance_to_divide.py

```diff
@@ -0,0 +1,234 @@
+    >>> elev = np.array([0.,  0.,  0., 0.,
+    ...                  0., 10., 10., 0.,
+    ...                  0., 20., 20., 0.,
+    ...                  0., 30., 30., 0.,
+    ...                  0.,  0.,  0., 0.])
+    >>> _ = mg.add_field('node','topographic__elevation', elev)
+    >>> mg.set_closed_boundaries_at_grid_edges(
+    ...     bottom_is_closed=False,
+    ...     left_is_closed=True,
+    ...     right_is_closed=True,
+    ...     top_is_closed=True)
+    >>> fr = FlowAccumulator(mg, flow_director = 'D8')
+    >>> fr.run_one_step()
+    >>> distance_to_divide = calculate_distance_to_divide(
+    ...     mg,
+    ...     add_to_grid=True,
+    ...     noclobber=False)
+    >>> mg.at_node['distance_to_divide']
+    array([ 0.,  3.,  3.,  0.,
+            0.,  2.,  2.,  0.,
+            0.,  1.,  1.,  0.,
+            0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.])
+
+    Now, let's change to MFD the flow_director method, which routes flow to
+    multiple nodes.
+
+    >>> from landlab import RasterModelGrid
+    >>> from landlab.components import FlowAccumulator
+    >>> from landlab.utils.distance_to_divide import (
+    ...     calculate_distance_to_divide)
+    >>> mg = RasterModelGrid((5, 4), xy_spacing=(1, 1))
+    >>> elev = np.array([0.,  0.,  0., 0.,
+    ...                  0., 10., 10., 0.,
+    ...                  0., 20., 20., 0.,
+    ...                  0., 30., 30., 0.,
+    ...                  0.,  0.,  0., 0.])
+    >>> _ = mg.add_field('node','topographic__elevation', elev)
+    >>> mg.set_closed_boundaries_at_grid_edges(
+    ...     bottom_is_closed=False,
+    ...     left_is_closed=True,
+    ...     right_is_closed=True,
+    ...     top_is_closed=True)
+    >>> fr = FlowAccumulator(mg, flow_director = 'MFD')
+    >>> fr.run_one_step()
+    >>> distance_to_divide = calculate_distance_to_divide(
+    ...     mg,
+    ...     add_to_grid=True,
+    ...     noclobber=False)
+    >>> mg.at_node['distance_to_divide']
+    array([ 0.,  3.,  3.,  0.,
+            0.,  2.,  2.,  0.,
+            0.,  1.,  1.,  0.,
+            0.,  0.,  0.,  0.,
+            0.,  0.,  0.,  0.])
+
+    The distance_to_divide utility can also work on irregular grids. For the
+    example we will use a Hexagonal Model Grid, a special type of Voroni Grid
+    that has regularly spaced hexagonal cells.
+
+    >>> from landlab import HexModelGrid, FIXED_VALUE_BOUNDARY, CLOSED_BOUNDARY
+    >>> from landlab.components import FlowAccumulator
+    >>> from landlab.utils.distance_to_divide import (
+    ...     calculate_distance_to_divide)
+    >>> dx = 1
+    >>> hmg = HexModelGrid(5,3, dx)
+    >>> _ = hmg.add_field('topographic__elevation',
+    ...                   hmg.node_x + np.round(hmg.node_y),
+    ...                   at = 'node')
+    >>> hmg.status_at_node[hmg.boundary_nodes] = CLOSED_BOUNDARY
+    >>> hmg.status_at_node[0] = FIXED_VALUE_BOUNDARY
+    >>> fr = FlowAccumulator(hmg, flow_director = 'D4')
```

Comment:
It's a bit odd that the name 'D4' applies here, though I understand why. For LL 2.0 we should change the name to, say, Dadjacent or something (or keep D4 for purposes of raster grids, but add Dadjacent as a synonym.)